### ISSUE TITLE
`PeerConnectionSpec`.`keep using transient connection` is flaky

### ISSUE THREAD

**Issue by pm47** [ACINQ/eclair#2276]
Introduced by #2096.

```
PeerConnectionSpec:

- keep using transient connection *** FAILED ***
  akka.actor.IllegalActorStateException: underlying actor is terminated
  at akka.actor.IllegalActorStateException$.apply(Actor.scala:161)
  at akka.testkit.TestActorRef.underlyingActor(TestActorRef.scala:119)
  at akka.testkit.TestFSMRef.fsm(TestFSMRef.scala:40)
  at akka.testkit.TestFSMRef.stateName(TestFSMRef.scala:45)
  at fr.acinq.eclair.io.PeerConnectionSpec.$anonfun$new$43(PeerConnectionSpec.scala:420)
```

The `peerConnection` actor gets killed due to: https://github.com/ACINQ/eclair/blob/8e2fc7acd35ef9c9c1e6318913f37ae225828c5a/eclair-core/src/main/scala/fr/acinq/eclair/io/PeerConnection.scala#L392-L398

I'm not sure how to fix, but the `sleep` method seems way too brittle for our test setup with lots of parallelization. Maybe some inspiration to take from here: https://doc.akka.io/docs/akka/current/typed/testing-async.html#controlling-the-scheduler

**Comment by t-bast** [comment]
I wasn't happy with the use of `sleep` here, I was pretty sure it would bite us with random test failures at some point...
Unfortunately I don't see a simple way of updating this test since our rate-limiter is based on timing.
One option I see would be to be able to inject a dummy, configurable rate-limiter in tests, but maybe it would be overkill.

**Comment by t-bast** [comment]
Fixed by #2277
